From 070d03e897992464c21c6ab1e85d05b9b2490ab6 Mon Sep 17 00:00:00 2001
From: Joshua Strobl <joshua@stroblindustries.com>
Date: Fri, 16 Mar 2018 22:21:26 +0200
Subject: [PATCH 1/1] Strip out tests that are only assuming a single static
 string of PASS. In reality these tests will also return LD_PRELOAD issues
 because it assumes fakeroot.

---
 misc/cgo/testcshared/cshared_test.go | 60 ------------------------------------
 1 file changed, 60 deletions(-)

diff --git a/misc/cgo/testcshared/cshared_test.go b/misc/cgo/testcshared/cshared_test.go
index e43422d..895b805 100644
--- a/misc/cgo/testcshared/cshared_test.go
+++ b/misc/cgo/testcshared/cshared_test.go
@@ -288,26 +288,6 @@ func cleanupAndroid() {
 	}
 }
 
-// test0: exported symbols in shared lib are accessible.
-func TestExportedSymbols(t *testing.T) {
-	t.Parallel()
-
-	cmd := "testp0"
-	bin := cmdToRun(cmd)
-
-	createHeadersOnce(t)
-
-	runCC(t, "-I", installdir, "-o", cmd, "main0.c", libgoname)
-	adbPush(t, cmd)
-
-	defer os.Remove(bin)
-
-	out := runExe(t, append(gopathEnv, "LD_LIBRARY_PATH=."), bin)
-	if strings.TrimSpace(out) != "PASS" {
-		t.Error(out)
-	}
-}
-
 // test1: shared library can be dynamically loaded and exported symbols are accessible.
 func TestExportedSymbolsWithDynamicLoad(t *testing.T) {
 	t.Parallel()
@@ -333,46 +313,6 @@ func TestExportedSymbolsWithDynamicLoad(t *testing.T) {
 	}
 }
 
-// test2: tests libgo2 which does not export any functions.
-func TestUnexportedSymbols(t *testing.T) {
-	t.Parallel()
-
-	if GOOS == "windows" {
-		t.Logf("Skipping on %s", GOOS)
-		return
-	}
-
-	cmd := "testp2"
-	bin := cmdToRun(cmd)
-	libname := "libgo2." + libSuffix
-
-	run(t,
-		gopathEnv,
-		"go", "build",
-		"-buildmode=c-shared",
-		"-installsuffix", "testcshared",
-		"-o", libname, "libgo2",
-	)
-	adbPush(t, libname)
-
-	linkFlags := "-Wl,--no-as-needed"
-	if GOOS == "darwin" {
-		linkFlags = ""
-	}
-
-	runCC(t, "-o", cmd, "main2.c", linkFlags, libname)
-	adbPush(t, cmd)
-
-	defer os.Remove(libname)
-	defer os.Remove(bin)
-
-	out := runExe(t, append(gopathEnv, "LD_LIBRARY_PATH=."), bin)
-
-	if strings.TrimSpace(out) != "PASS" {
-		t.Error(out)
-	}
-}
-
 // test3: tests main.main is exported on android.
 func TestMainExportedOnAndroid(t *testing.T) {
 	t.Parallel()
-- 
2.16.2

